### ISSUE TITLE
Provide means of disabling geolocation services that run on every boot

### ISSUE THREAD

**Issue by jksat** [OpenPLi/enigma2#2576]
As title states, users need means of disabling location check at every boot, this was created by @IanSav without any regard to end user choice.

This may be a sensitive matter to many and peolple start to question the entire E2 project, ([read comments](https://github.com/openatv/enigma2/issues/1563)) so, should be users that make the decision of using this services or not.

After all, this 'tool' is only needed by a very few users, so set for them timezone, for most there's no need to be running this at every boot.

**Comment by WanWizard** [comment]
I agree, it should not run on every boot, it should run in the setup wizard to pre-set the timezone.

/cc @IanSav @littlesat

**Comment by IanSav** [comment]
This matter has already been dealt with on OpenViX. Once I get confirmation that the code changes are working as intended the changes will be applied to OpenATV and OpenPLi.

See:
https://github.com/OpenViX/enigma2/commit/24a93fe29cd6005f5892ad346c6b758f8dca9d1a
https://github.com/OpenViX/enigma2/commit/65e04e583521f536c5031af8ab5682fa845f0bdf

I suggest you read the commit notes to understand why the code can't work the way we all wish it could.  For now I have made the best adjustments available.

There has been much discussion about this feature.  Most people are operating on rumours and innuendo and are being asked to voice objection based on incorrect information.  I would like to understand the basis of your objection to this code?

**Comment by WanWizard** [comment]
Sorry, but can you explain why it can't work the way we wish, i.e. default to some timezone in a new image, and have the setup wizard do a one-time check?

I have no issue with the alleged security implications, but merely with the fact that once a timezone is set, it is never needed again, and I don't like code that shouldn't be there.

**Comment by IanSav** [comment]
I will answer your question when I hear from this user.

**Comment by AbuBaniaz** [comment]
The code that has been in Vix/ATV/PLI for the last two months on every restart queries the server which sends back the co-ordinates of the user. 

The code never worked because if you flash PLI, you default timezone is Amsterdam. If you flash ATV, your default timezone is Berlin, if you flashed ViX, it was London. This has been fixed on Ian's last change.

The user, myself and others who are aware of this do not agree to our geographic co-ordinates being sent to/from receiver/server over an unsecure connection. No other consumer AV device that I know of requests a packet of data specifying its exact location on every restart. Most devices request permission before locating people/device and even then it is only to city.

What websites do with IP addresses is up to them. We are talking about Enigma2 code requesting a device's exact location. This is obviously based on IP address. We can play word games and say this is in the public domain. Addresses are public. So are names on certain versions of the electoral register unless they opt out. The register if births /deaths and marriages is also a public document. Enigma2 receivers are not built with security in mind.

Instead of dealing with the objection and making this function optional, people are being attacked and accused of instigating others to raise the issue. The person raising this issue has just created the GitHub account. Make of that what you will. It is no secret that Ian is not a friend of mine. However, there is no witchhunt as I am being accused of doing.

The timezone is only set once. Most people restore settings and don't even set it. For me, this function should be on user demand.  We can then talk about using the co-ordinates for setting USALS, langauge etc

The debate is about locating people/devices. Not whether IP addresses are public or not.

**Comment by WanWizard** [comment]
I don't debate any of that.

So: add the option to the setup wizard to retrieve it if the user wants to, or set it manually otherwise. And after that, don't bother, as you say a timezone is only set once.

**Comment by AbuBaniaz** [comment]
> So: add the option to the setup wizard to retrieve it if the user wants to, or set it manually otherwise. And after that, don't bother, as you say a timezone is only set once.

To me, that is the correct approach.

**Comment by Tony-il-Capo** [comment]
When I first raised some concerns about this (**more than 2 months ago**), I told IanSav that sooner or later this would create some mistrust, this can be a sensitive matter, and the way it was forced upon the user did not help.
Let's face it, very few actually needed this, but if someone actually need it, then make optional.

After endless discussions that were going nowhere, I let it go, then restored the old behavior for my setups.

Now, people started to notice the code and came here, they are now getting bullyed ("being asked", really??) because they just registered here to express themselves, I mean, you need an account to post, no?

Now, someone (on OpenViX) even labelled it as a "witch hunt"!

**Comment by IanSav** [comment]
I still haven't heard from the issue initiator.  When I hear from them I will respond in more detail.

For the pile on posts, please define *exactly* how geolocation is a cause for any issue.  Be sure you cover the fact that you were geolocated coming onto GitHub to post these complaints.  Also explain why you are  not complaining when you are being geolocated to do the DNS requests to actually get to GitHub.  Please also account for the fact that those geolocation lookups were sent to others and not your own machines like the Enigma2 geolocation lookup.

The fact is there are very few users who have an issue with the feature.  It is the same few vocal users that are posting.  They post in spite of the changes already made and being tested at the moment on OpenViX.

**Comment by AbuBaniaz** [comment]
Does GitHub sent a packet of data back to my browser/device with my exact locations based on the IP address?

You already posted the links to the ViX commits. WanWizard has already commented on them.

**Comment by Tony-il-Capo** [comment]
Jesus christ Ian!! Here we go again, I regret again coming back here.

Is this about DNS, GitHub accounts, web pages, or wtf is internet?
I don't think so.

> Provide means of disabling geolocation services that run on every boot

**Comment by IanSav** [comment]
No, GitHub logs it with Microsoft, the owner of GitHub.

The lookup for Enigma2 is never stored and never given to anyone other than the receiver that made the request.  The data is not stored there either.

**Comment by IanSav** [comment]
> Jesus christ Ian!! Here we go again, I regret again coming back here.

It is not your thread so coming here was your choice.

> Is this about DNS, GitHub accounts, web pages, or wtf is internet?
> I don't think so.

This statement shows a complete lack of knowledge about how the Internet works.  Geolocation is a very big part of Internet connectivity.

> > Provide means of disabling geolocation services that run on every boot

Done!  As mentioned in my first post in this thread.  Don't let facts get in the way of a good whinge.

**Comment by Tony-il-Capo** [comment]
Sure, keep insulting and judging people you know nothing about.
It's clear by now, that you feel like a superior being, everyone else is below, and either blind or stupid.

**Comment by IanSav** [comment]
Pot, meet kettle.

**Comment by AbuBaniaz** [comment]
I have submitted a pull request to exclude the IP address and co-ordinates from being sent over the net these are what some people are concerned about.

I don't know why Enigma2 needs to know who your ISP is nor what it will do with that data.

**Comment by Azureit** [comment]
@IanSav that openvix patch works for me, it's a way to disable geolocation.
But why not include also [my patch](https://github.com/openatv/enigma2/issues/1563#issuecomment-602054495)  to **initialize geolocation when needed**, the code was bad?

**Comment by IanSav** [comment]
The whole issue of geolocation has been made contentious due to a campaign of rumor and innuendo and harassment.  The fact that there was no compromise of security or identity didn't matter.  I was asked to make the code only run for this initial wizard detection and to not run again.  That is what I delivered.  Your code to limit the number of calls to the geolocation really didn't save much and did not comply with these requested changes.

It was generally agreed that the geolocation code was not a security risk but there was a concern that it was running unnecessarily.  The fact was that it only ever run once per boot session was never highlighted but always in the code.  To address the public panic that was stirred up the code now only runs to set the initial time zone selection in the wizard.  Any code to allow it to run more often would be contrary to the requested code change.
